commands.py

```diff
@@ -136,6 +136,43 @@ def handle_mode(state: mantatail.ServerState, user: mantatail.UserConnection, ar
         process_user_modes()
 
 
+def handle_nick(state: mantatail.ServerState, user: mantatail.UserConnection, args: List[str]) -> None:
+    """
+    Sets a user's nickname if they don't already have one.
+    Changes the user's nickname if they already have one.
+    """
+    nick_regex = r"[a-zA-Z|\\_\[\]{}^`-][a-zA-Z0-9|\\_\[\]{}^`-]{,15}"
+
+    if not args:
+        error_no_nickname_given(user)
+        return
+
+    new_nick = args[0]
+    if not re.match(nick_regex, new_nick):
```

Comment:
`re.match` is a terribly named function. For example, `re.match("hello", "hello world")` matches! A much better name for it would be `match_beginning` or `match_prefix` or something.

```suggestion
    if not re.fullmatch(nick_regex, new_nick):
```

Reply:
Oh, I thought `match` was a method in the `re` module. I didn't realize we could name these functions ourselves.

Reply:
It is in the `re` module, and you can't name it yourself. I'm saying that whoever created the `re` module named it badly.